src/module_4/groceries_model_fitting.md

```diff
@@ -0,0 +1,1642 @@
+train_cols = numerical_cols + binary_cols
+```
+
+# Base Model
+
+In order to know if we are following the right path, we will start by creating a simple model that will serve as a baseline and benchmark for future models.  we will take a simple predictor as the baseline(column: `Global popularity`). Since we are dealing with a binary classification problem (purchased or not purchased), the metrics used will be: Precision-Recall curve and AUC-ROC curve. The Precision-Recall curve it's a specially good metric to use when the classes are imbalanced, as is the case here. All these metrics are based on the famous confusion matrix.
+
+
+```python
+def calculate_metrics(
+    model_name: str,
+    train_proba: pd.Series,
+    y_train: pd.Series,
+    val_proba: pd.Series,
+    y_val: pd.Series,
+):
+    train_roc_auc = roc_auc_score(y_train, train_proba)
+    train_crossentropy = log_loss(y_train, train_proba)
+    train_ap = average_precision_score(y_train, train_proba)
+    val_roc_auc = roc_auc_score(y_val, val_proba)
+    val_crossentropy = log_loss(y_val, val_proba)
+    val_ap = average_precision_score(y_val, val_proba)
+
+    print(
+        f"{model_name}: Train AUC = {train_roc_auc:.3f}, Val AUC = {val_roc_auc:.3f}; "
+        f"Train AP = {train_ap:.3f}, Val AP = {val_ap:.3f}; "
+        f"Train Crossentropy = {train_crossentropy:.3f}, Val Crossentropy = {val_crossentropy:.3f}"
+    )
+    return train_ap, val_ap, train_crossentropy, val_crossentropy
+
+
+def plot_metrics(
+    model_name: str,
+    y_pred: pd.Series,
+    y_test: pd.Series,
+    figure: Tuple[matplotlib.figure.Figure, np.array] = None,
+):
+    precision_, recall_, _ = precision_recall_curve(y_test, y_pred)
+    pr_auc = auc(recall_, precision_)
+    fpr, tpr, _ = roc_curve(y_test, y_pred)
+    roc_auc = roc_auc_score(y_test, y_pred)
+    if figure is None:
+        fig, ax = plt.subplots(1, 2, figsize=(17, 4))
+    else:
+        fig, ax = figure
+
+    ax[0].plot(recall_, precision_, label=f"{model_name} (AUC = {pr_auc:.2f})")
+    ax[0].set_xlabel("Recall")
+    ax[0].set_ylabel("Precision")
+    ax[0].set_title("Precision-Recall curve")
+    ax[0].legend(loc="best")
+
+    ax[1].plot(fpr, tpr, label=f"{model_name} (AUC = {roc_auc:.2f})")
+    ax[1].set_xlabel("False Positive Rate")
+    ax[1].set_ylabel("True Positive Rate")
+    ax[1].set_title("ROC curve")
+    ax[1].legend(loc="best")
+```
+
+
+```python
+val_log_loss = []
+val_avg_precision = []
+calculate_metrics(
+    "Baseline model",
+    train_df["global_popularity"],
+    y_train,
+    val_df["global_popularity"],
+    y_val,
+)
+plot_metrics(
+    "Popularity baseline", y_pred=val_df["global_popularity"], y_test=val_df[target]
+)
+```
+
+    Baseline model: Train AUC = 0.787, Val AUC = 0.787; Train AP = 0.060, Val AP = 0.066; Train Crossentropy = 0.070, Val Crossentropy = 0.064
+
+
+
+    
+![png](groceries_model_fitting_files/groceries_model_fitting_15_1.png)
+    
+
+
```

Comment:
Up to here I assumed that it was the same as module 3, thus did not check thoroughly and all the comments on module 3 applies